tests/Unit/EnclosureTest.php

```diff
@@ -58,4 +60,135 @@ public function testClassExists()
     {
         $this->assertTrue(class_exists('SimplePie_Enclosure'));
     }
+
+    /**
+     * @dataProvider getLinkProvider
+     */
+    public function test_get_link($data, $expected)
+    {
+        $feed = new SimplePie();
+        $feed->set_raw_data($data);
+        $feed->enable_cache(false);
+        $feed->init();
+
+        $item = $feed->get_item(0);
+        $enclosure = $item->get_enclosure(0);
+
+        $this->assertInstanceOf(Enclosure::class, $enclosure);
+        $this->assertSame($expected, $enclosure->get_link());
+    }
+
+    public function getLinkProvider()
+    {
+        return [
+            'Test enclosure get_link urlencoded' => [
+<<<EOT
+<rss version="2.0" xmlns:media="http://search.yahoo.com/mrss/">
+    <channel>
+        <title>Test enclosure link 1</title>
+        <description>Test enclosure link 1</description>
+        <link>http://example.net/tests/</link>
+        <item>
+            <title>Test enclosure link 1.1</title>
+            <description>Test enclosure link 1.1</description>
+            <guid>http://example.net/tests/#1.1</guid>
+            <link>http://example.net/tests/#1.1</link>
+            <media:content url="http://example.net/link?a=%22b%22&amp;c=%3Cd%3E" medium="image">
+            </media:content>
+        </item>
+    </channel>
+</rss>
+EOT
+            ,
+                'http://example.net/link?a=%22b%22&amp;c=%3Cd%3E',
+            ],
+            'Test enclosure get_link urldecoded' => [
+<<<EOT
+<rss version="2.0" xmlns:media="http://search.yahoo.com/mrss/">
+    <channel>
+        <title>Test enclosure link 2</title>
+        <description>Test enclosure link 2</description>
+        <link>http://example.net/tests/</link>
+        <item>
+            <title>Test enclosure link 2.1</title>
+            <description>Test enclosure link 2.1</description>
+            <guid>http://example.net/tests/#2.1</guid>
+            <link>http://example.net/tests/#2.1</link>
+            <media:content url="http://example.net/link?a=&quot;b&quot;&amp;c=&lt;d&gt;" medium="image">
+            </media:content>
+        </item>
+    </channel>
+</rss>
+EOT
+            ,
+                'http://example.net/link?a=%22b%22&amp;c=%3Cd%3E',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider getEnclosuresProvider
+     */
+    public function test_get_enclosures($data, $expected)
+    {
+        $feed = new SimplePie();
+        $feed->set_raw_data($data);
+        $feed->enable_cache(false);
+        $feed->init();
+
+        $item = $feed->get_item(0);
+        $enclosures = $item->get_enclosures();
+
+        $this->assertSame($expected, count($enclosures));
```

Comment:
What do you think about using `assetCount()`?

```suggestion
        $this->assertCount($expected, $item->get_enclosures());
```